script.embuary.info/resources/lib/tmdb_main.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/python
+# coding: utf-8
+
+########################
+
+import sys
+import xbmc
+import xbmcgui
+
+from resources.lib.helper import *
+from resources.lib.tmdb_utils import *
+from resources.lib.tmdb_person import *
+from resources.lib.tmdb_video import *
+
+########################
+
+class TheMovieDB(object):
+    def __init__(self,call,params):
+        self.window_stack = []
+        self.dialog_cache = {}
+        self.call = call
+        self.tmdb_id = params.get('tmdb_id')
+        self.query = remove_quotes(params.get('query'))
+        self.query_year = params.get('year')
+        self.external_id = params.get('external_id')
+
+        busydialog()
```

Comment:
An example of why the context manager would serve better here. You have a few examples on the forum

Reply:
What do you mean with context manager?

Reply:
https://forum.kodi.tv/showthread.php?tid=333950&pid=2754914#pid2754914

Reply:
Ah ok. Wont work in my case because there can be select dialogs between and I don't want to show the busy dialog in this case. I've checked all scenarios more than twice. Busydialognocancel geta always closed. But I agree that it's the better approach in a more common and simple case.

Reply:
Ok.